gtsam/hybrid/tests/testIncrementalHybrid.cpp

```diff
@@ -45,296 +44,334 @@ using symbol_shorthand::X;
-//   for (size_t i = 0; i <= 4; i++) {
-//     graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(i));
-//   }
-
-//   // Create ordering.
-//   Ordering ordering;
-//   for (size_t j = 1; j <= 4; j++) {
-//     ordering += X(j);
-//   }
-
-//   size_t maxComponents = 5;
-//   incrementalHybrid.update(graph1, ordering, maxComponents);
-
-//   auto actualBayesNet1 = incrementalHybrid.hybridBayesNet();
-//   CHECK_EQUAL(4, actualBayesNet1.size());
-//   EXPECT_LONGS_EQUAL(2, actualBayesNet1.atGaussian(0)->nrComponents());
-//   EXPECT_LONGS_EQUAL(4, actualBayesNet1.atGaussian(1)->nrComponents());
-//   EXPECT_LONGS_EQUAL(8, actualBayesNet1.atGaussian(2)->nrComponents());
-//   EXPECT_LONGS_EQUAL(5, actualBayesNet1.atGaussian(3)->nrComponents());
-
-//   GaussianHybridFactorGraph graph2;
-//   graph2.push_back(switching.linearizedFactorGraph.dcGraph().at(3));
-//   graph2.push_back(switching.linearizedFactorGraph.gaussianGraph().at(5));
-
-//   Ordering ordering2;
-//   ordering2 += X(4);
-//   ordering2 += X(5);
-
-//   incrementalHybrid.update(graph2, ordering2, maxComponents);
-
-//   auto actualBayesNet = incrementalHybrid.hybridBayesNet();
-//   CHECK_EQUAL(2, actualBayesNet.size());
-//   EXPECT_LONGS_EQUAL(10, actualBayesNet.atGaussian(0)->nrComponents());
-//   EXPECT_LONGS_EQUAL(5, actualBayesNet.atGaussian(1)->nrComponents());
-// }
+/* ****************************************************************************/
+// Test if we can incrementally do the inference
+TEST(DCGaussianElimination, Incremental_inference) {
+  Switching switching(3);
+  IncrementalHybrid incrementalHybrid;
+  GaussianHybridFactorGraph graph1;
+
+  // Create initial factor graph
+  // *- X1 -*-  X2 -*-X3
+  //     \*-M1-*/
+  graph1.push_back(switching.linearizedFactorGraph.dcGraph().at(0));
+  graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(0));
+  graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(1));
+  graph1.push_back(switching.linearizedFactorGraph.gaussianGraph().at(2));
+
+  // Create ordering.
+  Ordering ordering;
+  ordering += X(1);
+  ordering += X(2);
 
-/* ************************************************************************* */
-/* Test for figuring out the optimal ordering to ensure we get a discrete graph
- * after elimination. */
+  // Run update step
+  incrementalHybrid.update(graph1, ordering);
+
+  // Test if hybrid bayes net is the correct size and if the keys are as
+  // expected.
+  auto hybridBayesNet = incrementalHybrid.hybridBayesNet();
+  EXPECT_LONGS_EQUAL(2, hybridBayesNet.size());
+  EXPECT(hybridBayesNet.at(0)->frontals() == KeyVector{X(1)});
+  EXPECT(hybridBayesNet.at(0)->parents() == KeyVector({X(2), M(1)}));
+  EXPECT(hybridBayesNet.at(1)->frontals() == KeyVector{X(2)});
+  EXPECT(hybridBayesNet.at(1)->parents() == KeyVector({M(1)}));
+
+  // Check if the remaining factor graph and DecisionTreeFactor of discrete
```

Comment:
// Check that the remaining factor graph has a single decision tree factor on M1.